hammett/conf/__init__.py

```diff
@@ -9,30 +9,184 @@
 See the global_settings.py for a list of all possible settings.
 """
 
+import copy
 import importlib
+import operator
 import os
 
 from hammett.conf import global_settings
 from hammett.core.exceptions import ImproperlyConfigured
 from hammett.core.hiders import HidersChecker
 
+_EMPTY = object()
+
 _HAMMETT_SETTINGS_MODULE = 'HAMMETT_SETTINGS_MODULE'
 
 
+def new_method_proxy(func):
```

Comment:
It's preferable to use [wraps](https://docs.python.org/3/library/functools.html#functools.wraps)

Reply:
In most cases, yes, but not here. I left a comment with the explanation.